Sources/Common/Core/DataArray/index.js

```diff
@@ -425,8 +425,23 @@ function vtkDataArray(publicAPI, model) {
   };
 
   publicAPI.deepCopy = (other) => {
+    // Retain current dataType and array reference before shallowCopy call.
+    const currentType = publicAPI.getDataType();
+    const currentArray = model.values;
     publicAPI.shallowCopy(other);
-    publicAPI.setData(other.getData().slice());
+
+    // Avoid array reallocation if size already sufficient
+    // and dataTypes match.
```

Comment:
Are you trying to retain the original array pointer here?

Reply:
Yes. In case the size of arrays is the same or the destination array is large enough, I am trying to avoid an unnecessary new allocation of array. From the point of view of any pipeline (that uses deep copy of DataArrays), and might get updated frequently due to state changes (e.g. mouse drag operation).